### DISCUSSION TITLE
How to recover an instance when the hostagent dies?

### DISCUSSION THREAD

**Discussion by jandubois** [lima-vm/lima#390] [Q&A] [answered]
When the hostagent process is dead, there is no way to connect to the instance, or to stop it:

```console
$ limactl ls
WARN[0000] instance "default" has errors                 errors="[qemu is running but host agent is not]"
NAME       STATUS     SSH                ARCH       CPUS    MEMORY    DISK      DIR
default    Broken     127.0.0.1:0        x86_64     4       4GiB      100GiB    /Users/jan/.lima/default

$ lima
Bad port '0'

$ limactl stop
FATA[0000] expected status "Running", got "Broken"
```

You can of course `limactl delete -f default`, but then all the data on disk is gone too.

The only way to regain access to the instance is by killing the qemu process:

```
pkill -F ~/.lima/default/qemu.pid
```

Or of course by rebooting the host, which will kill all running VMs too.

Should there be a better way? Or at least more helpful output than "expected Running, got Broken"?

PS: This is somewhat hypothetical, as I normally don't see the hostagent dying; for this test I killed it manually.

**Comment by AkihiroSuda** [comment, marked as answer]
`stop -f` works?
